### PR TITLE
8304585: Method::invoke rewraps InvocationTargetException if a caller-sensitive method throws IAE

### PR DESCRIPTION
**Pull request by mlchung**
A simple fix to `Method::invoke` which wraps IAE with `InvocationTargetException` twice if it's thrown by a caller-sensitive method which has no adapter.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8304585](https://bugs.openjdk.org/browse/JDK-8304585): Method::invoke rewraps InvocationTargetException if a caller-sensitive method throws IAE

### Reviewers
 * [Joe Darcy](https://openjdk.org/census#darcy) (@jddarcy - **Reviewer**) ⚠️ Review applies to [45c7daf0](https://git.openjdk.org/jdk/pull/13233/files/45c7daf0554720e1f0caff61568f2887def8fc8f)
 * [Jaikiran Pai](https://openjdk.org/census#jpai) (@jaikiran - **Reviewer**)
 * [Alan Bateman](https://openjdk.org/census#alanb) (@AlanBateman - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/13233/head:pull/13233` \
`$ git checkout pull/13233`

Update a local copy of the PR: \
`$ git checkout pull/13233` \
`$ git pull https://git.openjdk.org/jdk.git pull/13233/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 13233`

View PR using the GUI difftool: \
`$ git pr show -t 13233`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/13233.diff">https://git.openjdk.org/jdk/pull/13233.diff</a>

</details>

### CODE REVIEW

**Review by jddarcy — APPROVED** [review]
Looks fine.

File: test/jdk/java/lang/reflect/Method/CallerSensitiveMethodInvoke.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+     * should throw InvocationTargetException thrown by Field::get.
+     */
+    @Test
+    public void csMethodInvoke() throws ReflectiveOperationException {
+        Field f = CallerSensitiveMethodInvoke.class.getDeclaredField("value");
+        try {
+            // Field::get throws IAE
+            Method m = Field.class.getDeclaredMethod("get", Object.class);
+            m.invoke(f, new Object());
+        } catch (InvocationTargetException e) {
```
Comment (liach) [inline test/jdk/java/lang/reflect/Method/CallerSensitiveMethodInvoke.java:56]: The test should fail if invoke executes without throwing, same for below

Comment (mlchung) [reply test/jdk/java/lang/reflect/Method/CallerSensitiveMethodInvoke.java:56]: not following what you mean.  Can you elaborate?

Comment (liach) [reply test/jdk/java/lang/reflect/Method/CallerSensitiveMethodInvoke.java:56]: Though very unlikely, I recommend adding a line of `Assertions.ail("Should not reach here")` after the `m.invoke` and before the catch to ensure the invocation always throw `InvocationTargetException`.

Comment (mlchung) [reply test/jdk/java/lang/reflect/Method/CallerSensitiveMethodInvoke.java:56]: I see what you mean.  They should always throw.  I could add assert.

**Review by jaikiran — APPROVED** [review]
Looks fine to me.

**Review by AlanBateman — APPROVED** [review]
The removal of the wrapping looks okay to me.

**Pull request closed without merging**